src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -671,11 +671,11 @@ int32_t EditorRemoveUnusedObjects()
             {
                 const ObjectRepositoryItem* item = &items[i];
                 ObjectType objectType = item->Type;
+                if (ObjectTypeIsIntransient(objectType))
+                    continue;
 
-                if (objectType >= ObjectType::SceneryGroup)
-                {
+                if (objectType == ObjectType::ParkEntrance || objectType == ObjectType::Water)
```

Comment:
whats the reason behind these two?

Reply:
I believe they are the only types where the game allows one and only one object selected.

Reply:
Ah yeah good point. Can you add a comment stating that reason.